include/rocksdb/options.h

```diff
@@ -492,6 +492,25 @@ struct DBOptions {
   // Default: false
   bool track_and_verify_wals_in_manifest = false;
 
+  // EXPERIMENTAL: This API/behavior is subject to change
+  // If true, during DB-open try verifying the SST unique id between MANIFEST
+  // and SST properties, which is to make sure the SST is not overwritten or
+  // misplaced. A corruption error will be reported if mismatch detected, but
+  // only when MANIFEST tracks the unique id, which starts from version 7.3.
+  // The user could also the unique id from API
+  // `GetUniqueIdFromTableProperties()`.
+  //
+  // Note:
+  // 1. the option should be used with `max_open_files=-1` to check all SST
+  //    files. Otherwise, only the opened files during DB-open are checked.
+  // 2. existing SST files won't have its unique_id tracked in MANIFEST, then
+  //    verification will be skipped.
+  // To check how many SST verifications are skipped, a statistic ticker
+  // `NUMBER_SST_UNIQUE_ID_VERIFICATION_SKIP` is introduced for that.
+  //
+  // Default: false
+  bool try_verify_sst_unique_id = false;
```

Comment:
Why not drop the "try" prefix? Though we skip verification in specific conditions, I am not sure whether we should have try prefix...

Reply:
I think this is a good point. After a few releases, most SSTs would have the unique_id tracking in the MANIFEST, so the try_ in the name will become anachronistic.

Reply:
This option is probably most closely analogous is probably `track_and_verify_wals_in_manifest` though `track_and_verify_sst_unique_ids_in_manifest` is probably too verbose.

The reason for having an option controlling some enhancement is because it incurs some trade-off in performance or correctness. I believe there's more correctness risk in *not* verifying, and it's not clear to me that the performance cost of tracking IDs in the manifest is measurable. Also consider that the (small) extra memory footprint is present regardless of enablement, so it would be misleading to suggest that disabling recovers all lost performance. And AFAIK there's no forward compatibility concern. So I'm not sure what the option is really for.

Reply:
In short, having too many options is both a performance and usability issue.

Reply:
> Why not drop the "try" prefix? Though we skip verification in specific conditions, I am not sure whether we should have try prefix...
sounds good, we can remove "try_"

Reply:
we changed the feature to verify all SST files, regardless of `max_open_files` setting. Now there is performance overhead of enabling the feature, which reads all SST files properties in DB-open, so no plan to enable the feature by default for now (maybe adding a feature to only verify opened file without exposing any interface).